datavault-webapp/src/main/java/org/datavaultplatform/webapp/controllers/admin/AdminRolesController.java

```diff
@@ -0,0 +1,307 @@
+
+    }
+
+
+    private void createNewRole(String name, String type, String description, String[] permissions)
+    {
+
+        RoleModel role = new RoleModel();
+
+        List<PermissionModel> pms = getSelectedPermissions(type, permissions);
+
+        role.setPermissions(pms);
+        role.setName(name);
+        role.setDescription(description);
+        role.setType(RoleType.valueOf(type));
+
+        restService.createRole(role);
+
+    }
+
+    private List<PermissionModel> getSelectedPermissions(String type, String[] permIds )
+    {
+        List<PermissionModel> pms = getPermissionsByType(type);
+
+        ArrayList<PermissionModel> result = new ArrayList<>();
+
+        if (pms == null) return
+                new ArrayList<>();
+
+        for (String id : permIds)
+        {
+            Optional<PermissionModel> optional = pms.stream().filter(x -> x.getId().equals(id)).findFirst();
+            PermissionModel perm = optional.isPresent() ? optional.get() : null;
+
+            if (perm != null)
+                result.add(perm);
+
+        }
+          return result;
+    }
+
+    private List<PermissionModel> getPermissionsByType(String type) {
+
+        List<PermissionModel> model;
+
+        switch (type.toUpperCase()) {
+            case "VAULT":
+                model = restService.getVaultPermissions();
+                break;
+            case "SCHOOL":
+                model = restService.getSchoolPermissions();
+                break;
+            default:
+                model = null;
+
+        }
+
+        return model;
+    }
+
+
+
+    @RequestMapping(value = "/admin/roles/delete/{id}", method = {RequestMethod.GET, RequestMethod.POST})
+    public String deleteRole(ModelMap model, @PathVariable("id") long roleid, RedirectAttributes redirectAttributes) throws Exception {
+
+        try {
+
+            logger.info("Role has been requested to be deleted id:" + roleid);
+
+            RoleModel role = getRoleById(roleid);
+
+            if (role == null) {
+                model.addAttribute("errormessage", "Roles does not exist Role Id:" + roleid);
+                 setCommonModel(model);
+                 return "/admin/roles/index";
+
+            } else {
+
+                if (role.getAssignedUserCount() > 0) {
+                    model.addAttribute("errormessage", "Can't delete a role that has users");
+                    setCommonModel(model);
+                    return "/admin/roles/index";
+                } else {
+                    //attempt to delete role
+                    restService.deleteRole(roleid);
+                }
+            }
+
+
+        } catch (Exception e) {
+            String msg = "Error trying to delete  role " + roleid + " Error " + e.getMessage();
+            logger.error(msg, e);
+
+            model.addAttribute("errormessage", msg);
+
+            setCommonModel(model);
+            return "/admin/roles/index";
+
+        }
+            return "redirect:/admin/roles/";
```

Comment:
Needs consistent formatting, looks like a few other places might be affected with a cursory glance.